ovh/resource_cloud_project_user.go

```diff
@@ -149,6 +150,50 @@ func validateCloudProjectUserRoleFunc(v interface{}, k string) (ws []string, err
 	return
 }
 
+func resourceCloudProjectUserUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	userId := d.Id()
+	role := d.Get("role_name")
+	roles, _ := helpers.StringsFromSchema(d, "role_names")
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/role",
+		url.PathEscape(serviceName),
+	)
+	res := &CloudProjectrolesResponse{}
+	if err := config.OVHClient.Get(endpoint, res); err != nil {
+		return fmt.Errorf("calling Get %s", endpoint)
+	}
+
+	update := []string{}
+	for _, i := range res.Roles {
+		for _, j := range roles {
+			if j == i.Name {
+				update = append(update, i.Id)
+			}
+		}
+		if role == i.Name {
```

Comment:
what if the role is also in `role_names` ? It will be twice in the array ? Shouldn't you `break` in the if above ?